javascript/javascript2/week2/currencyCnvrtrWeek2.js

```diff
@@ -0,0 +1,132 @@
+// currencyConverter.js
+
+const exRates = {
+  timestamp: Date.now(),
+  base: "USD",
+  date: new Date().toISOString().split("T")[0],
+  exRates: {
+    USD: 1.23396,
+    GBP: 0.88204,
+    EUR: 0.92398,
+    CAD: 1.36325,
+  },
+};
+
+const addNewRateform = document.getElementById("addNewRateform");
+const convertCurrencyForm = document.getElementById("convertCurrencyForm");
+const updateRateForm = document.getElementById("updateRateForm");
+const addRateResult = document.getElementById("addRateResult");
+const convertCurrencyResult = document.getElementById("convertCurrencyResult");
+const updateRateResult = document.getElementById("updateRateResult");
+const searchRateResult = document.getElementById("searchRateResult");
+
+// Search function for currency:
+function onRateSearch(event) {
+  event.preventDefault();
+  const searchFrom = document
+    .getElementById("search-from")
+    .value.trim()
+    .toUpperCase();
+  const searchTo = document
+    .getElementById("search-to")
+    .value.trim()
+    .toUpperCase();
+
+  if (exRates.base === searchFrom && exRates.exRates[searchTo]) {
```

Comment:
Be careful, the brief says `searching by the currency from OR to`.

Reply:
Hi vdaguenet, That means, I need to change the AND operator to OR operator in the line number 35 right?

Reply:
It will be a bit more complex, but first you need to add the array of currency rate objects as described here https://github.com/Rajesh-Bhatt/hyf-homework/pull/6#discussion_r1675882053. Then you will create a search function that will filter the array. Here is the expected result of the search function in pseudo code:

```js
function searchRate(baseCurrency, toCurrency) {
  // performs searching and returns a currency rate object
}

// Search the rate where the base is EUR and has a rate value for DKK
searchRate('EUR', 'DKK'); // returns { base: 'EUR', rates: { 'DKK': 123 } }
// Search all the rates that have a rate value for DKK
searchRate(null, 'DKK'); // returns [{ base: 'EUR', rates: { 'DKK': 123 } }, { base: 'USD', rates: { 'DKK': 234 } }]
// Search the rate where the base is USD
searchRate('USD', null); //  return { base: 'USD', rates: { 'DKK': 234 } }
```